pmd-core/src/main/java/net/sourceforge/pmd/ant/Formatter.java

```diff
@@ -57,12 +61,26 @@ public Renderer getRenderer() {
     }
 
     public void start(String baseDir) {
+
+        Properties properties = createProperties();
+        
+        Charset charset;
+        {
+            String s = (String) properties.get("encoding");
+            if (null == s) {
+                charset = StandardCharsets.UTF_8;
+                properties.put("encoding", charset.name());
+            } else {
+                charset = Charset.forName(s);
+            }
+        }
+
         try {
             if (toConsole) {
-                writer = new BufferedWriter(new OutputStreamWriter(System.out));
+                writer = new BufferedWriter(new OutputStreamWriter(System.out, charset));
```

Comment:
I can confirm the `sun.stdout.encoding` property is present in Oracle JRE and OpenJDK, but not on other JREs such as IBM's J9. However, the reflection based approach used by JMH does work on all 3 JREs, and most probably all others, since it's based on the API exposed in class sources that are freely available.

I'd suggest we adopt the reflection detection code for console encoding from JMH for maximum compatibility and leave the rest as is.